build.sh

```diff
@@ -2,53 +2,200 @@
+    
+    # Check if packages are already installed
+    for package in $APK_PACKAGES; do
+        if ! apk info -e "$package" &> /dev/null; then
+            echo_warning "Package $package is not installed. Installing..."
+            # Install silently, redirecting output to /dev/null, but keep errors
+            if sudo apk add "$package" &> /dev/null; then
+                echo_plain "Package $package installed successfully."
+            else
+                echo_error "Failed to install $package."
+            fi
+        else
+            echo_plain "$package is already installed."
+        fi
+    done
+}
+
+# Function to install packages via apt (with silent installation)
+install_via_apt() {
+    echo_plain "Detected apt"
+    
+    # Check if packages are already installed
+    for package in $APT_PACKAGES; do
+        if ! dpkg -l | grep -q "$package"; then
+            echo_warning "Package $package is not installed. Installing..."
+            # Install silently, redirecting output to /dev/null, but keep errors
+            if sudo apt install -y "$package" &> /dev/null; then
+                echo_plain "Package $package installed successfully."
+            else
+                echo_error "Failed to install $package."
+            fi
+        else
+            echo_plain "$package is already installed."
+        fi
+    done
+}
+
+# Function to install packages via pacman (with silent installation)
+install_via_pacman() {
+    echo_plain "Detected pacman"
+    
+    # Check if packages are already installed
+    for package in $PACMAN_PACKAGES; do
+        if ! pacman -Qi "$package" &> /dev/null; then
+            echo_warning "Package $package is not installed. Installing..."
+            # Install silently, redirecting output to /dev/null, but keep errors
+            if sudo pacman -S --noconfirm "$package" &> /dev/null; then
+                echo_plain "Package $package installed successfully."
+            else
+                echo_error "Failed to install $package."
+            fi
+        else
+            echo_plain "$package is already installed."
+        fi
+    done
+}
+
+
+# Function to install dependencies for most Linux systems
+install_dependencies_for_most_linux_systems() {
+  echo_success "Checking dependencies"
+
   if command -v apk &> /dev/null; then
-      echo "Detected apk"
-      sudo apk add python3 py3-pip py3-virtualenv
+      install_via_apk
   elif command -v apt &> /dev/null; then
-      echo "Detected apt"
-      sudo apt install python3-pip python3-virtualenv
+      install_via_apt
   elif command -v pacman &> /dev/null; then
-      echo "Detected pacman"
-      sudo pacman -S --noconfirm python-pip python-virtualenv
+      install_via_pacman
   else
-      echo "Unsupported package manager. Please install the packages and create the virtualenv manually."
+      echo_error "Unsupported package manager. Please install the packages and create the virtualenv manually."
```

Comment:
The way it's implemented right now with `exit 1`, it's misleading to say users should install dependecies, since even if they did, it would still error out. Better to be factual and just say "Failed to detect package manager"

If the exit would be removed, the script could continue. In that case this should be a warning about package manager detection. 

In no case does it make sense to instruct about virtualenv, since this would be a separate subsequent step.